Rock.Messaging.RabbitMQ/RabbitEngine.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Threading.Tasks;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Rock.Messaging.RabbitMQ
+{
+    public class RabbitEngine : IReceiver, ISender
+    {
+        private readonly IRabbitSessionConfiguration config;
+        private ConnectionFactory connectionFactory;
+        private IConnection connection;
+        private List<IModel> consumerModels; // While this is hacky- IModel is not thread safe.. see: https://www.rabbitmq.com/releases/rabbitmq-dotnet-client/v1.7.1/rabbitmq-dotnet-client-1.7.1-user-guide.pdf
+
+        private bool topical;
+
+        public RabbitEngine(IRabbitSessionConfiguration rabbitSessionConfiguration, bool isTopic = false)
+        {
+            this.config = rabbitSessionConfiguration;
+            topical = isTopic;
+            consumerModels = new List<IModel>();
+            Initialize();
+        }
+        private void Initialize()
+        {
+            var connectionStrs = config.QueueUrl.Split(':');
+            connectionFactory = new ConnectionFactory
+            {
+                UserName = config.UserName,
+                Password = config.Password,
+                HostName = connectionStrs[0],
+                Port = int.Parse(connectionStrs[1]),
+                VirtualHost = config.vHost,
+            };
+            connection = connectionFactory.CreateConnection();
+            
+        }
+
+        public void Dispose()
+        {
+            connection.Dispose();
+            foreach (var m in consumerModels)
+            {
+                m.Abort();
+            }
+            
+        }
+
+        #region Receiver
+        public void Start(string selector) // TODO: Discuss: RabbitMQ supports getting individual messages- should Rock.Messaging support that?
+        {
+            var model = connection.CreateModel();
```

Comment:
Sure does, only in the Receiver instance, though.... and I believe IReceiver/ISender are IDisposable.